orcid_service/views.py

```diff
@@ -32,10 +32,16 @@ def get_access_token():
     # do not use connection pool, always establish a new connection to the orcid remote server
     # we were having issue with dropped connectins mid-stream and this request is not idempotent
     # therefore we can't retry
-    r = requests.post(current_app.config['ORCID_OAUTH_ENDPOINT'], data=data, headers=headers)
+    try:
+        r = requests.post(current_app.config['ORCID_OAUTH_ENDPOINT'], data=data, headers=headers)
+    except requests.exceptions.ConnectionError:
```

Comment:
You may want to add other exceptions (see [example](https://github.com/adsabs/ADSCore/blob/2da27b82528d58fb385da8830bfbd1bfd76638f3/adscore/api.py#L290)) such as `ConnectTimeout` and `ReadTimeout`. At this point, no need to have a custom message for each type of exception.

And also, you could add a timeout to the requests with an argument `timeout=30` so that you are sure that it won't stay blocked for a long time (otherwise, the timeout depends on their server side).